install.sh

```diff
@@ -21,6 +21,13 @@ gpasswd -a ${SUDO_USER:-$USER} epoptes
 echo 'IPAPPEND=3' >> /etc/ltsp/update-kernels.conf
 /usr/share/ltsp/update-kernels
 
+# Configuring resolvconf
+echo "nameserver 8.8.8.8
```

Comment:
That's right. When you get an IP from a DHCP server, you also get DNS servers as well (if the DHCP server is configured correctly). So, in general there is no need to set the nameservers manually (unless there is some other problem in your configurations).

Reply:
I don't think there is any problem with my configuration. `dnsmasq` is not properly configured with linux mint 19. As soon as it installs it gives an error message 
```
    default: Setting up dnsmasq (2.79-1) ...
    default: Created symlink /etc/systemd/system/multi-user.target.wants/dnsmasq.service → /lib/systemd/system/dnsmasq.service.
    default: Job for dnsmasq.service failed because the control process exited with error code.
    default: See "systemctl status dnsmasq.service" and "journalctl -xe" for details.
    default: invoke-rc.d: initscript dnsmasq, action "start" failed.
    default: [0;1;31m●[0m dnsmasq.service - dnsmasq - A lightweight DHCP and caching DNS server
    default:    Loaded: loaded (/lib/systemd/system/dnsmasq.service; enabled; vendor preset: enabled)
    default:    Active: [0;1;31mfailed[0m (Result: exit-code) since Fri 2018-07-13 15:16:33 IST; 14ms ago
    default:   Process: 7603 ExecStart=/etc/init.d/dnsmasq systemd-exec [0;1;31m(code=exited, status=2)[0m
    default:   Process: 7602 ExecStartPre=/usr/sbin/dnsmasq --test (code=exited, status=0/SUCCESS)
    default: Jul 13 15:16:33 vagrant-mint17-xfce32 systemd[1]: Starting dnsmasq - A lightweight DHCP and caching DNS server...
    default: Jul 13 15:16:33 vagrant-mint17-xfce32 dnsmasq[7602]: dnsmasq: syntax check OK.
    default: Jul 13 15:16:33 vagrant-mint17-xfce32 dnsmasq[7603]: dnsmasq: failed to create listening socket for port 53: Address already in use
    default: Jul 13 15:16:33 vagrant-mint17-xfce32 dnsmasq[7603]: [0;1;31m[0;1;39m[0;1;31mfailed to create listening socket for port 53: Address already in use[0m
    default: Jul 13 15:16:33 vagrant-mint17-xfce32 dnsmasq[7603]: [0;1;31m[0;1;39m[0;1;31mFAILED to start up[0m
    default: Jul 13 15:16:33 vagrant-mint17-xfce32 systemd[1]: [0;1;39m[0;1;31m[0;1;39mdnsmasq.service: Control process exited, code=exited status=2[0m
    default: Jul 13 15:16:33 vagrant-mint17-xfce32 systemd[1]: [0;1;39m[0;1;31m[0;1;39mdnsmasq.service: Failed with result 'exit-code'.[0m
    default: Jul 13 15:16:33 vagrant-mint17-xfce32 systemd[1]: [0;1;31m[0;1;39m[0;1;31mFailed to start dnsmasq - A lightweight DHCP and caching DNS server.[0m
```
Thats why we have we have to switch to 5353 and configure nameserver manually.
`auto_config` is a good idea. But we cannot use in case of linux mint 19

Reply:
Deepanshu, LinuxMint19 works OK out of the box, without installing other things on it. So don't blame it if something does not work. Instead try to find what creates the problem, and then fix it.

In particular, ltsp tries to install and configure `dnsmaq`, while `ubuntu` is already using one (because LinuxMint is actually Ubuntu). So, try to fix the dnsmasq configuration of ltsp so that it does not conflict with the existing one. I have also shown you previously some references about this.

Reply:
I made a fresh install of linux mint 19. Did apt update and then installed dnsmasq.
It does not work - https://youtu.be/afV2g3Af5ikix 
(That too on actual virtual machine not the vagrant box)

It gives error message right after installation. Meaning I did not make any configuration mistake. 

In order to find the problem I did a side by side comparision of linux mint 19 and ubuntu bionic.
downloaded dnsmasq package using apt. Checked their hash value. Both were same
But when I installed it in `bionic` it worked and when I installed it in `tara` it didn't.
Error message was port 53 is already being used.
Then I did a `ss -ant` to check what ports are both machines listening to.
Both were listening to port 53 yet one is working other does not. That points something is wrong with `tara`.
`systemd-resolved` works well with dnsmasq in `bionic` but not with `tara`. I tried
> I have also shown you previously some references about this.

Yes I did check that one. They are switching to google DNS. At this steps i just frezees
`nmcli connection down "$NetManProfile"`
Switching dnsmasq to a different port is a good idea. It was hinted by @akash0x53

Reply:
@d78ui98 there is a simple logic here. Port 53 is used for DNS service, but the `dnsmasq` that you are installing does not need to offer DNS service, so it does not need to use port 53.

The reference that I was talking about is this one: http://dashohoxha.fs.al/ltsp-pnp/#sec-3-4
It does mention a potential problem with DNS and its solution (or maybe another similar solution may be used).

Reply:
Yes, I did that. But it did not work. After googling about it for a while I had to ask on irc on how to solve this. They said it is a know issue, you can use the new ltsp to solve it. Doing that will also solve some other networking problems that one can face when using ltsp with ubuntu 18.

I implemented `auto_config`. I followed their guide and booted from the client. The on the client worked. That suggests it was ltsp problem not in our configuration.

I think we should switch to new ltsp. As it build for using ltsp on ubuntu 18.

Reply:
> I think we should switch to new ltsp. As it build for using ltsp on ubuntu 18.

Great. If the new ltsp works well, then we should definitely switch to it.